src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.15;
+
+import "./ZkBobPool.sol";
+
+/**
+ * @title ZkBobCompoundingMixin
+ */
+abstract contract ZkBobCompoundingMixin is ZkBobPool {
+    using SafeERC20 for IERC20;
+
+    uint256 public investedAssetsAmount;
+
+    struct YieldParams {
+        // ERC4626 vault address (or address(0) if not set)
+        address yield;
+        // maximum amount of underlying tokens that can be invested into vault
+        uint256 maxInvestedAmount;
+        // expected amount of underlying tokens to be left at the pool after successful rebalance
+        uint256 buffer;
+        // slippage/rounding protection buffer, small part of accumulated interest that is non-claimable
+        uint96 dust;
+        // address to receive accumulated interest during the rebalance
+        address interestReceiver;
+        // operator address (or address(0) if permissionless)
+        address yieldOperator;
+    }
+
+    YieldParams public yieldParams;
+
+    event UpdateYieldParams(YieldParams yieldParams);
+    event Claimed(address indexed token, address indexed yield, uint256 amount);
```

Comment:
Do we really need a token address here, if it's assumed to be constant over time?